fe/fe-core/src/main/java/com/starrocks/backup/RestoreJob.java

```diff
@@ -832,6 +869,60 @@ protected void sendCreateReplicaTasks() {
         }
     }
 
+    protected void addRestoreOlapView(Map<View, Boolean> restoredOlapViews) {
+        Database db = globalStateMgr.getLocalMetastore().getDb(dbId);
+        for (Map.Entry<View, Boolean> entry : restoredOlapViews.entrySet()) {
+            View restoredOlapView = entry.getKey();
+            boolean existed = entry.getValue();
+
+            if (existed) {
+                // already existed, need to alter the view
+                AlterViewClause alterViewClause = new AlterViewClause(null,
+                                    restoredOlapView.getQueryStatement(), NodePosition.ZERO);
+                alterViewClause.setInlineViewDef(restoredOlapView.getInlineViewDef());
+                alterViewClause.setColumns(restoredOlapView.getColumns());
+                alterViewClause.setComment(restoredOlapView.getComment());
+                AlterViewStmt alterViewStmt = new AlterViewStmt(
+                                    new TableName(db.getFullName(), restoredOlapView.getName()),
+                                    alterViewClause, NodePosition.ZERO);
+
+                ConnectContext context = new ConnectContext();
+                context.setDatabase(db.getFullName());
+                context.setGlobalStateMgr(globalStateMgr);
+                context.setStartTime();
+                context.setThreadLocalInfo();
+
+                new AlterJobExecutor().process(alterViewStmt, context);
+                LOG.info("replace view {} successfully in restore", restoredOlapView.getName());
+            } else {
+                List<Column> columns = restoredOlapView.getColumns();
+                long tableId = globalStateMgr.getNextId();
+                String viewName = restoredOlapView.getName();
+                View view = new View(tableId, viewName, columns);
+                view.setComment(restoredOlapView.getComment());
+                view.setInlineViewDefWithSqlMode(restoredOlapView.getInlineViewDef(),
+                                                 restoredOlapView.getSqlMode());
+                // init here in case the stmt string from view.toSql() has some syntax error.
+                try {
+                    view.init();
+                } catch (Exception e) {
+                    status = new Status(ErrCode.COMMON_ERROR, "failed to init View in restore" +
+                                        restoredOlapView.getName() + " " + e.getMessage());
+                    return;
+                }
+
+                try {
+                    globalStateMgr.getLocalMetastore().onCreate(db, view, "", false);
+                } catch (DdlException e) {
+                    status = new Status(ErrCode.COMMON_ERROR, "failed to create View in restore" +
```

Comment:
better to use `LocalMetastore.createView()` to create or alter view.
need refactor `LocalMetastore.createView()` first to reuse the code.